ot/torch/lp.py

```diff
@@ -79,3 +78,69 @@ def ot_solve(a, b, M, num_iter_max=100000, log=False):
         G = emd(a2, b2, M2, log=False, numItermax=num_iter_max)
 
         return torch.from_numpy(G).type_as(M)
+
+
+def emd1D_loss(u_values, v_values, u_weights=None, v_weights=None, p=1, require_sort=True):
+    r"""
+    Computes the 1 dimensional earth moving distance between two empirical distributions
+    ..math:
+        EMD &= \int_0^1 |cdf_u^{-1}(q)  cdf_v^{-1}(q)|^p dq
+
+    We do so in a vectorized way by first building the individual quantile functions then integrating them.
+    This has a theoretically higher complexity than the core OT implementation but behaves better with PyTorch
+
+    Parameters
+    ----------
+    u_values: torch.Tensor (..., n)
+        locations of the first empirical distribution
+    v_values: torch.Tensor (..., m)
+        locations of the second empirical distribution
+    u_weights: torch.Tensor (..., n), optional
```

Comment:
this should be a size n only vector if im not mistaken having the same weights for a given sample makes sens anyways

Reply:
Not necessarily, I'm planning ahead for sliced barycenter: in this case you have two batched dimensions for the locations and 1 for the weights

Reply:
Though arguably I haven't tested it yet but that would have happened gradually